src/pages/events/create.svelte

```diff
@@ -0,0 +1,162 @@
+<script>
+  import Uploader from '../../components/Uploader.svelte'
+  import { me } from '../../data/user'
+  import { events, create } from '../../data/events'
+  import { format, addDays } from 'date-fns'
+  import LocationInput from '../../components/LocationInput.svelte'
+  import Icon from 'fa-svelte'
+  import { throwError } from '../../data/error'
+  import { goto } from '@roxi/routify'
+  
+  let imageUrl = ''
+  
+  const defaults = {
+    name: '',
+    start_date: new Date(),
+    end_date: new Date(),
+  }
+  let event = {}
+  const tomorrow = format(addDays(Date.now(), 1), 'yyyy-MM-dd')
+            
+  function assertHas(value, errorMessage) {
+    if (!value) {
+      throwError(errorMessage)
+    }
+  }
+
+  function validate(event) {
+    assertHas(event.name, 'Please provide the name of the event')
+    assertHas(event.location, 'Please provide a location')
+    assertHas(event.start_date, 'Please tell us the start date')
+    assertHas(event.end_date, 'Please tell us the end date')
+  }
+  
+  async function onSubmit() {
+    validate(event)
+  
+    await create({
```

Comment:
For consistency (and to keep this file a little cleaner) move the object into the `create` function in events.js so this looks like:
```
await create(event)
```
You can look at the `update` function in requests.js as an example if this doesn't make sense.